### ISSUE TITLE
Rule proposal: useless override

### ISSUE THREAD

**Issue by DetachHead** [typescript-eslint/typescript-eslint#4747]
### Description

when a method is overridden but all it does is call the superclass's method, there should be a warning stating that the override is useless.

### Fail

```ts
class Foo {
    bar(): void {}
}

class Bar extends Foo {
    override bar(): void {
        super.bar()
    }
}
```

### Pass

```ts
class Foo {
    bar(): void {}
}

class Bar extends Foo {
    override bar(): void {
        console.log("doing something else")
        super.bar()
    }
}
```

**Comment by bradzacher** [comment]
Does this specifically need to be a TS rule?
It seems like there's no specific need to check for the `override` keyword - just look for any class method/class property arrow function which only includes a call to `super.<name>()`.

It feels like something that could be build directly into the core ESLint ruleset to benefit everyone?
https://eslint.org/docs/rules/no-useless-constructor

**Comment by DetachHead** [comment]
https://github.com/eslint/eslint/issues/15737

**Comment by DetachHead** [comment]
i think there's room for typescript-specific functionality however, see https://github.com/eslint/eslint/issues/15737#issuecomment-1081345873

**Comment by DetachHead** [comment]
> But wait... The examples in initial post. In both cases `override` is required. @DetachHead So the `override` is not useless. And this is indeed handles by TypeScript and not ESLint.

how so? in the first example all `Bar.bar` does is call the original method, so the functionality is exactly the same if the overridden method was removed

**Comment by DetachHead** [comment]
> Why would you declare such useless method?

the same question could be asked about empty functions. one reason is that IDEs usually have autocomplete which makes it much easier to accidentally leave useless functions like this lying around:

![](https://github.com/typescript-eslint/typescript-eslint/assets/57028336/3dab32a6-194a-4a19-a7bd-755e9aaa8ada)

**Comment by kirkwaiblinger** [comment]
I'm getting lost reading through the conversation here. Is there anything actionable beyond reporting 

```ts
class A extends OtherClass {
    uselessOverride(a: string) {
         return super.uselessOverride(a);
    }
}
```

(whether or not the `override` keyword is actually present)

**Comment by DetachHead** [comment]
been a while but i believe that's the conclusion we came to

**Comment by kirkwaiblinger** [comment]
Understood, thanks for the quick response.

We've decided we're going to close this issue since it's 
1. not typescript-specific functionality, nor is the rule significantly improved by using type information
2. been open for a long time and garnered very few community interactions
3. easy to implement as a custom rule

Thanks for discussion!
